modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/Netty4Transport.java

```diff
@@ -244,6 +251,37 @@ protected Netty4TcpServerChannel bind(String name, InetSocketAddress address) {
         return esChannel;
     }
 
+    @Override
+    protected MemoryUsage memoryStats() {
+        MemoryUsage baseMemoryUsage = super.memoryStats();
+
+        ByteBufAllocator allocator = ByteBufAllocator.DEFAULT;
```

Comment:
This is more of a detail, but don't we also have to grab the stats off of `io.netty.buffer.UnpooledByteBufAllocator#DEFAULT` if we have the pooled allocator running by default? Otherwise we miss all the heap usage our REST layer creates in the transport stats since it uses `Unpooled.allocate` right?

Reply:
👍

Reply:
We never release the bytes used at the REST level so those metrics would increase forever. In fact those metrics would throw off the network level metrics in this PR currently, so I will probably need to separate the allocators (if we are using unpooled). I will look into that.

Reply:
Anyway I guess what I'm saying here is that I view application level bytes (in the rest request and transport deserialized messages) are probably beyond the scope of this PR currently as it would involve adding a lifecycle to these messages.